rts/Sim/Units/Scripts/CobThread.cpp

```diff
@@ -49,14 +46,24 @@ CR_REG_METADATA_SUB(CCobThread, CallInfo,(
 	CR_MEMBER(stackTop)
 ))
 
+std::vector<decltype(CCobThread::dataStack)> CCobThread::freeDataStacks;
+std::vector<decltype(CCobThread::callStack)> CCobThread::freeCallStacks;
 
 CCobThread::CCobThread(CCobInstance* _cobInst)
 	: cobInst(_cobInst)
 	, cobFile(_cobInst->cobFile)
 {
+	if (!freeDataStacks.empty()) {
+		assert(freeDataStacks.size() == freeCallStacks.size());
+		dataStack = std::move(freeDataStacks.back());
+		freeDataStacks.pop_back();
+		callStack = std::move(freeCallStacks.back());
+		freeCallStacks.pop_back();
+	} else {
+		dataStack.reserve(32);
+		callStack.reserve(4);
```

Comment:
How arbitrary/empirical are these? Consider:
```suggestion
		/* 4 calls involving 32 bytes of data is a typical/generous/eyeballed/??
		 * amount, as measured across Recoil/BAR/lol actually just janus/??. */
		callStack.reserve(4);
```

Reply:
Let's avoid mentioning games in the engine source code.

Reply:
What would be a good case to profile on other games (ZK/MF/TAP)?

Reply:
I wouldn't mind profiling on other scenarios, I only require instructions somewhere on the best flags to compile for it.

Reply:
I think mentioning a game is fine if it is a clear reference point for a measurement. ZK only uses BOS/COB scripting for chickens which might use a somewhat above-average amount of data, idk.

Reply:
Assume any code you write might be (un)read(able) 5-10 years down the line.

Reply:
That is fine. If somebody sees "as measured for BAR janus" and nobody knows what a bar janus is anymore, then that's a good sign the measurement is no longer useful. Or if people know what a bar janus is, but in 10+ years games all have super duper inverse kinematics for most of their units, then that's also a good sign the measurement is obsolete. If it's just "as measured for a typical unit in a typical major game" that's technically true, but also leaves you little room to say "this needs to be revised" because a typical unit in a typical game is perpetually the target, just it's not the same unit and not the same game.

Reply:
Anyway, I'm up for profiling what would be a good benchmark for another game (and being unbiased towards BAR). If one such scenario is provided as well as a set of compilation flags for tracy I can perform it, I believe it's worth assessing if this patch provides performance improvements with no cons.

Reply:
I will prepare plots with distribution of sizes, it doesn't matter that much IMHO, but interesting to take a look at.

Reply:
Yes, the stack sizes converge to the max over time due to pooling. Initial size is only to avoid the cost of initial doubling. Whatever all stacks converge depends a lot of distribution + length of the game.

Reply:
Good, I was initially worried that there would be perpetual resizing if the default was small. In that case consider:
```suggestion
		/* Arbitrary defaults; most stacks in the pool will eventually get
		 * assigned to large scripts and grow to accomodate them, at some
		 * point the whole pool will converge to that size */
		dataStack.reserve(32);
		callStack.reserve(4);
```